.github/workflows/check-pull-request.yml

```diff
@@ -40,47 +37,106 @@ jobs:
       - run: cargo +nightly fmt --all -- --check
 
   lint:
-    if: ${{ !startsWith(github.head_ref, 'release/') }}
     name: Clippy
     runs-on: ubuntu-latest
     continue-on-error: false
     steps:
       - uses: actions/checkout@v4
       - run: ./.github/scripts/free_disk_space.sh
       - run: sudo apt-get install -y protobuf-compiler
-      - uses: actions/cache@v4
+
+      - name: Fetch cache
+        uses: Swatinem/rust-cache@a95ba195448af2da9b00fb742d14ffaaf3c21f43 # v2.7.0
+        with:
+          # consistent cache across jobs
+          shared-key: "bajun-cache-cargo-debug"
+
+      - run: cargo clippy --workspace --all-features --all-targets -- -D warnings
+        env:
+          SKIP_WASM_BUILD: 1
+
+  cargo-zepter:
+    name: Cargo Zepter
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Install stable Rust
+        uses: actions-rs/toolchain@v1
         with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-          key: ${{ runner.os }}-cargo-lint-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: ${{ runner.os }}-cargo-lint-
-      - run: cargo clippy --workspace --all-features --all-targets ${{ env.CARGO_ARGS }} -- -D warnings
+          profile: minimal
+          toolchain: stable
+
+      - name: Install Zepter
+        run: cargo install --locked -q zepter && zepter --version
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # Don't clone historic commits.
+
+      - name: Check features
+        run: zepter run check
 
   test:
-    if: ${{ !startsWith(github.head_ref, 'release/') }}
-    name: Test
+    name: Cargo test (all features)
     runs-on: ubuntu-latest
     continue-on-error: false
-    strategy:
-      matrix:
-        runtime:
-          - bajun-runtime
```

Comment:
I removed the runtime from the test, I think we always want to run all tests in the repo anyhow.